src/omv/modules/py_image.c

```diff
@@ -1106,145 +1066,126 @@ static mp_obj_t py_image_to(pixformat_t pixfmt, const uint16_t *default_color_pa
-                imlib_draw_image(&temp, src_img, 0, 0, arg_x_scale, arg_y_scale, &arg_roi,
-                                 arg_rgb_channel, arg_alpha, color_palette, alpha_palette,
-                                 (hint & (~IMAGE_HINT_CENTER)) | IMAGE_HINT_BLACK_BACKGROUND, NULL, NULL, NULL);
+                imlib_draw_image(&temp, src_img, 0, 0, x_scale, y_scale, &roi,
+                                 args[ARG_channel].u_int, args[ARG_alpha].u_int, color_palette, alpha_palette,
+                                 args[ARG_hint].u_int, NULL, NULL, NULL);
             }
 
-            if (((dst_img.pixfmt == PIXFORMAT_JPEG) && jpeg_compress(&temp, &dst_img_tmp, arg_q, false, subsampling))
+            if (((dst_img.pixfmt == PIXFORMAT_JPEG) &&
+                 jpeg_compress(&temp, &dst_img_tmp, args[ARG_quality].u_int, false, args[ARG_subsampling].u_int))
                 || ((dst_img.pixfmt == PIXFORMAT_PNG) && png_compress(&temp, &dst_img_tmp))) {
                 mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("Compression Failed!"));
             }
-        } else if (arg_e) {
+        } else if (args[ARG_encode_for_ide].u_bool) {
             dst_img_tmp.data = fb_alloc(image_size(&dst_img_tmp), FB_ALLOC_NO_HINT);
             memcpy(dst_img_tmp.data, src_img->data, dst_img_tmp.size);
         } else {
             dst_img_tmp.data = src_img->data;
         }
 
-        if (arg_e) {
+        if (args[ARG_encode_for_ide].u_bool) {
             dst_img.size = fb_encode_for_ide_new_size(&dst_img_tmp);
         } else {
             dst_img.size = dst_img_tmp.size;
         }
     }
 
+    // Determine the destination buffer.
+    // copy_to_fb == True (copy() method)
+    //  - args[ARG_copy].u_bool == True: Use the frame buffer.
+    //  - args[ARG_copy].u_bool == False: Allocate a new buffer.
+    // copy_to_fb == False (all other methods)
+    //  - args[ARG_copy].u_bool == True: Allocate a new buffer.
+    //  - args[ARG_copy].u_bool == False: Use the source buffer.
+
     uint32_t size = image_size(&dst_img);
 
-    if (copy) {
+    if (args[ARG_copy].u_bool) {
         if (copy_to_fb) {
             py_helper_set_to_framebuffer(&dst_img);
         } else {
             dst_img.data = xalloc(size);
         }
-    } else if (arg_other) {
-        bool fb = py_helper_is_equal_to_framebuffer(arg_other);
-        size_t buf_size = fb ? framebuffer_get_buffer_size() : image_size(arg_other);
-        PY_ASSERT_TRUE_MSG((size <= buf_size),
-                           "The new image won't fit in the target frame buffer!");
-        // DO NOT MODIFY arg_other YET (as it could point to src_img)!
-        dst_img.data = arg_other->data;
-        // Update now if not in place to possibly freeup framebuffer RAM, otherwise update after.
-        if (dst_img.data != src_img->data) {
-            py_helper_update_framebuffer(&dst_img);
-        }
+    } else if (!copy_to_fb) {
+        bool fb = py_helper_is_equal_to_framebuffer(src_img);
```

Comment:
I noticed that this function might not be safe to use anymore. The FB head can change before/after this function is called. It worked fine for a while, way before the multiple framebuffers were introduced but probably not anymore.

Reply:
The head would change after setting the frame buffers back to 1 from 3. If you force 1 frame buffer first is should be the same.